pulsar-flink/src/main/java/org/apache/flink/batch/connectors/pulsar/BasePulsarOutputFormat.java

```diff
@@ -85,7 +90,8 @@ public void configure(Configuration configuration) {
 
     @Override
     public void open(int taskNumber, int numTasks) throws IOException {
-        this.producer = getProducerInstance();
+        producer = getProducerInstance();
+        producerRefCnt.incrementAndGet();
```

Comment:
How about move this ref change into the `synchronized` block of `getProducerInstance`?